third_party/litex-donut/donut.c

```diff
@@ -21,18 +22,25 @@
 
 static signed char b[1760], z[1760];
 
+#ifdef OPT_LINK_CODE_IN_SRAM
+void donut(void) __attribute__((section(".ramtext")));
+#else
 void donut(void);
+#endif
+
+
 void donut(void) {
   int sA=1024,cA=0,sB=1024,cB=0,_;
+  int R1 = 1, R2 = 2048, K2 = 5120*1024;
   puts("\nPress any key to exit.\n");
   for (;;) {
+    unsigned start_cycle = perf_get_mcycle();
     memset(b, 32, 1760);  // text buffer
     memset(z, 127, 1760);   // z buffer
     int sj=0, cj=1024;
     for (int j = 0; j < 90; j++) {
       int si = 0, ci = 1024;  // sine and cosine of angle i
       for (int i = 0; i < 324; i++) {
-        int R1 = 1, R2 = 2048, K2 = 5120*1024;
```

Comment:
Why did you move this line?

Reply:
I moved it back to avoid confusion (also in the accelerated version).